qa/L0_input_validation/input_validation_test.py

```diff
@@ -116,101 +118,113 @@ def test_input_validation_all_optional(self):
 
 
 class InputShapeTest(unittest.TestCase):
-    def test_input_shape_validation(self):
-        input_size = 8
-        model_name = "pt_identity"
-        triton_client = tritongrpcclient.InferenceServerClient("localhost:8001")
+    def test_client_input_shape_validation(self):
+        model_name = "simple"
 
-        # Pass
-        input_data = np.arange(input_size)[None].astype(np.float32)
-        inputs = [
-            tritongrpcclient.InferInput(
-                "INPUT0", input_data.shape, np_to_triton_dtype(input_data.dtype)
-            )
-        ]
-        inputs[0].set_data_from_numpy(input_data)
-        triton_client.infer(model_name=model_name, inputs=inputs)
-
-        # Larger input byte size than expected
-        input_data = np.arange(input_size + 2)[None].astype(np.float32)
-        inputs = [
-            tritongrpcclient.InferInput(
-                "INPUT0", input_data.shape, np_to_triton_dtype(input_data.dtype)
-            )
-        ]
-        inputs[0].set_data_from_numpy(input_data)
-        # Compromised input shape
-        inputs[0].set_shape((1, input_size))
-        with self.assertRaises(InferenceServerException) as e:
-            triton_client.infer(
-                model_name=model_name,
-                inputs=inputs,
+        for client_type in ["http", "grpc"]:
+            if client_type == "http":
+                triton_client = tritonhttpclient.InferenceServerClient("localhost:8000")
+            else:
+                triton_client = tritongrpcclient.InferenceServerClient("localhost:8001")
+
+            # Infer
+            inputs = []
+            if client_type == "http":
+                inputs.append(tritonhttpclient.InferInput("INPUT0", [1, 16], "INT32"))
+                inputs.append(tritonhttpclient.InferInput("INPUT1", [1, 16], "INT32"))
+            else:
+                inputs.append(tritongrpcclient.InferInput("INPUT0", [1, 16], "INT32"))
+                inputs.append(tritongrpcclient.InferInput("INPUT1", [1, 16], "INT32"))
+
+            # Create the data for the two input tensors. Initialize the first
+            # to unique integers and the second to all ones.
+            input0_data = np.arange(start=0, stop=16, dtype=np.int32)
+            input0_data = np.expand_dims(input0_data, axis=0)
+            input1_data = np.ones(shape=(1, 16), dtype=np.int32)
+
+            # Initialize the data
+            inputs[0].set_data_from_numpy(input0_data)
+            inputs[1].set_data_from_numpy(input1_data)
+
+            # Compromised input shapes
+            inputs[0].set_shape([2, 8])
+            inputs[1].set_shape([2, 8])
+
+            with self.assertRaises(InferenceServerException) as e:
+                triton_client.infer(model_name=model_name, inputs=inputs)
+            err_str = str(e.exception)
+            self.assertIn(
+                f"unexpected shape for input 'INPUT1' for model 'simple'. Expected [-1,16], got [2,8]",
```

Comment:
Is this a client-side error or core-side error? Client PR only refers to "unexpected element count" errors from what I can see.

If this is a core error message, please rename test to `test_core_input_validation` or something

Reply:
I would expect `1, 16` -> `2. 8` to pass an "element count" check where both have a product of 16, but fail a shape check.

Reply:
This is the error from core (client does not know model_name). This test checks
1. request is not caught by the client check in the case of [1, 16] -> [2, 8].
2. request is caught by the client check in the case of [1, 16] -> [1, 8].

Reply:
Right, so if we're not testing client-side validation in this function, then let's rename `test_client_input_shape_validation` to `test_core_input_shape_validation`

Reply:
Both checks are in `test_client_input_shape_validation` test case in order to share the same input. Do you want me to split them into two tests cases?

Reply:
Let's just add a comment above the first check: https://github.com/triton-inference-server/server/pull/7427/files#r1705894309

Reply:
Like it.